### PR TITLE
[RW-2441][risk=moderate] Only init Zendesk Widget when user is signed in

### PR DESCRIPTION
**Pull request by jmthibault79**
Potential security risk: imports external executable code (no change from current)

### CODE REVIEW

File: ui/src/app/utils/zendesk.ts [hunk 1]
```diff
@@ -1,3 +1,49 @@
...
+  const s = document.createElement('script');
+  s.type = 'text/javascript';
+  s.id = 'ze-snippet';
+  s.src = 'https://static.zdassets.com/ekr/snippet.js?key=' + accessKey;
+  elementRef.nativeElement.appendChild(s);
+
+  // This data configures the Zendesk web widget with settings to show only
+  // the "contact us" form. See https://developer.zendesk.com/embeddables/docs/widget/
+  // for API docs.
+  window['zESettings'] = {
```
Comment (jmthibault79) [inline ui/src/app/utils/zendesk.ts]: This is purely cut/paste from index.html

**Comment by jmthibault79** [comment]
The JIRA ticket suggests moving the `<script>` calls to `signed-in/component.html`.  This was not possible however, because Angular ignores script tags in templates.  I used the solution given in https://stackoverflow.com/questions/38088996/adding-script-tags-in-angular-component-template instead.

File: ui/src/app/utils/zendesk.ts [hunk 2]
```diff
@@ -1,3 +1,49 @@
...
+export function initializeZendeskWidget(
+  elementRef: ElementRef, accessKey: string): void {
+  // This external script loads the Zendesk web widget, connected to our
+  // production Zendesk account. If we ever get a test Zendesk account, the key
+  // should be templated in using an environment variable.
+  const s = document.createElement('script');
+  s.type = 'text/javascript';
+  s.id = 'ze-snippet';
+  s.src = 'https://static.zdassets.com/ekr/snippet.js?key=' + accessKey;
+  elementRef.nativeElement.appendChild(s);
```
Comment (jmthibault79) [inline ui/src/app/utils/zendesk.ts]: This is intended to be equivalent to
```
<script id="ze-snippet" src="https://static.zdassets.com/ekr/snippet.js?key=5a7d70b9-37f9-443b-8d0e-c3bd3c2a55e3"> </script>
```

**Comment by blrubenstein** [comment]
Taking a look now. (Also, FYI @ericsong I use they/them pronouns)

File: ui/src/app/utils/zendesk.ts [hunk 3]
```diff
@@ -1,3 +1,49 @@
...
+
+  // This data configures the Zendesk web widget with settings to show only
```
Comment (blrubenstein) [inline ui/src/app/utils/zendesk.ts]: Nit: I'd throw a comment on here about why it needs to be done this way.

File: ui/src/app/utils/zendesk.ts [hunk 4]
```diff
@@ -1,3 +1,53 @@
+import {ElementRef} from '@angular/core';
+
+// ideally this initialization would be done by including <script> tags in
```
Comment (jmthibault79) [inline ui/src/app/utils/zendesk.ts]: @blrubenstein how is this?

**Pull request merged**